specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json

```diff
@@ -0,0 +1,2571 @@
+          "enum": [
+            "S1"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IotDpsSku",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "OperationInputs": {
+      "description": "Input values for operation results call.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the Provisioning Service to check.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "NameAvailabilityInfo": {
+      "type": "object",
+      "description": "Description of name availability.",
+      "properties": {
+        "nameAvailable": {
+          "description": "specifies if a name is available or not",
+          "type": "boolean"
+        },
+        "reason": {
+          "description": "specifies the reason a name is unavailable",
+          "enum": [
+            "Invalid",
+            "AlreadyExists"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "NameUnavailabilityReason",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "description": "message containing a detailed reason name is unavailable",
+          "type": "string"
+        }
+      }
+    },
+    "TagsResource": {
+      "properties": {
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "A container holding only the Tags for a resource, allowing the user to update the tags on a Provisioning Service instance."
+    },
+    "ProvisioningServiceDescriptionListResult": {
+      "type": "object",
+      "description": "List of provisioning service descriptions.",
+      "properties": {
+        "value": {
+          "description": "List of provisioning service descriptions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ProvisioningServiceDescription"
+          }
+        },
+        "nextLink": {
+          "description": "the next link",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "VerificationCodeResponse": {
+      "type": "object",
+      "description": "Description of the response of the verification code.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of certificate.",
+          "readOnly": true
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Request etag."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
```

Comment:
Refactoring:

IF this is a resource type... THEN you should inherit allOf ProxyResource from common-types here also. This should be nonbreaking.

BUT... seems like you only actually return this from a POST operation?
In that case it isn't actually a resource, its just a HTTP response, so its not required to have 'name', 'id', 'type' properties like resources are when you GET them.

Reply:
I would suggest to remove all of the properties here that you are returning which are not actually used.

Reply:
You also don't need the 'properties' wrapper for POST responses.